parquet-column/src/main/java/parquet/example/data/simple/convert/SimplePrimitiveConverter.java

```diff
@@ -18,13 +18,18 @@
  */
 package parquet.example.data.simple.convert;
 
+import parquet.column.Dictionary;
+import parquet.example.data.simple.Primitive;
+import parquet.io.ParquetDecodingException;
 import parquet.io.api.Binary;
 import parquet.io.api.PrimitiveConverter;
+import parquet.io.api.RecordConsumer;
 
 class SimplePrimitiveConverter extends PrimitiveConverter {
 
   private final SimpleGroupConverter parent;
   private final int index;
+  private DictionaryPrimitive[] dictionaryPrimitives;
```

Comment:
I have simplified this code and removed DictionaryPrimitive class. I think I still need a factory method to create correct type of Primitive (this is why I added type info to Dictionary). I have moved the factory method to Primitive. Let me know if you think there is a preferred approach.